nexus_constructor/instrument.py

```diff
@@ -0,0 +1,59 @@
+import os
+import h5py
+from nexus_constructor.component_type import make_dictionary_of_class_definitions
+from nexus_constructor.nexus import nexus_wrapper as nx
+from nexus_constructor.component import ComponentModel
+
+COMPONENTS_IN_ENTRY = ["NXmonitor", "NXsample"]
+
+
+def convert_name_with_spaces(component_name):
+    return component_name.replace(" ", "_")
+
+
+class Instrument:
+    def __init__(self, nexus_file: nx.NexusWrapper):
+        self.nexus = nexus_file
+        _, self.nx_component_classes = make_dictionary_of_class_definitions(
+            os.path.abspath(
+                os.path.join(
+                    os.path.realpath(__file__), os.pardir, os.pardir, "definitions"
+                )
+            )
+        )
+
+    def add_component(self, name: str, nx_class: str, description: str):
+        """
+        Creates a component group in a NeXus file
+        :param name: Name of the component group to create
+        :param nx_class: NX_class of the component group to create
+        :param description: Description of the component
+        :return Wrapper for added component
+        """
+        name = convert_name_with_spaces(name)
+        parent_group = self.nexus.instrument
+        if nx_class in COMPONENTS_IN_ENTRY:
+            parent_group = self.nexus.entry
+        component_group = self.nexus.create_nx_group(name, nx_class, parent_group)
+        component = ComponentModel(self.nexus, component_group)
+        component.description = description
+        return component
+
+    def remove_component(self, component: ComponentModel):
+        """
+        Removes a component group from the NeXus file
+        :param component: The component to be removed
+        """
+        self.nexus.delete_group(component.group)
+
+    def get_component_list(self):
+        component_list = []
```

Comment:
Similar to previous comment, it would be nice if this list is updated instead of re-created.